File: BEAKJOON/C++/BOJ5032.cpp
#include <iostream>

int main(){
    int e,f,c;
    std::cin >> e >> f >> c;
    int sum = e+f;
    int tot = 0;
    
    while (sum >= c)
    {
        tot += sum/c;
        sum = sum/c + sum%c;
    }
    std::cout << tot;
}

File: BEAKJOON/Java/BOJ5032.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class BOJ5032 {
    public static void main(String[] args) throws IOException{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        int e = Integer.parseInt(st.nextToken());
        int f = Integer.parseInt(st.nextToken());
        int c = Integer.parseInt(st.nextToken());

        int sum = e+f;
        int tot = 0;

        while (sum >= c) {
            tot += sum/c;
            sum = sum/c + sum%c;
        }

        System.out.println(tot);
    }
}
